pkg/config/config_test.go

```diff
@@ -479,3 +482,46 @@ func TestConfig_GetYamlAttribute(t *testing.T) {
 	assert.True(t, c.ConnectEnabled)
 	assert.Error(t, c.SetBoolValueByYamlAttribute("no_a_value", false))
 }
+
+func (s *ConfigSuite) Test_ParseIncludeMatchingRules(c *C) {
+	config := `
+license_key: test
+include_matching_metrics:
+  process.name:
+    - test
+    - other-test
+  process.executable:
+    - regex "^some-process" 
+`
+	f, err := ioutil.TempFile("", "include_matching_rules_config_test")
+	c.Assert(err, IsNil)
+	_, err = f.WriteString(config)
+	c.Assert(err, IsNil)
+	_ = f.Close()
+	defer func(f *os.File) {
+		err = os.Remove(f.Name())
+		c.Assert(err, IsNil)
+	}(f)
+
+	cfg, err := LoadConfig(f.Name())
+	c.Assert(err, IsNil)
+	c.Assert(cfg.IncludeMetricsMatchers, HasLen, 2)
+	c.Assert(cfg.IncludeMetricsMatchers["process.name"], HasLen, 2)
+	c.Assert(cfg.IncludeMetricsMatchers["process.executable"], HasLen, 1)
+}
+
+func (s *ConfigSuite) Test_ParseIncludeMatchingRule_EnvVar(c *C) {
+	os.Setenv("NRIA_INCLUDE_MATCHING_METRICS", "process.name:\n - regex \"kube*\" \n")
+	defer os.Unsetenv("NRIA_INCLUDE_MATCHING_METRICS")
+
+	configStr := "license_key: abc123"
+	f, err := ioutil.TempFile("", "yaml_config_test")
+	c.Assert(err, IsNil)
+	f.WriteString(configStr)
+	f.Close()
+
+	cfg, err := LoadConfig(f.Name())
+	c.Assert(err, IsNil)
+	c.Assert(fmt.Sprintf("%v", cfg.IncludeMetricsMatchers), Equals, "map[process.name:[regex \"kube*\"]]")
```

Comment:
good for now but this assertion would become flaky if a new rule is added as Golang maps don't provide field order guarantees